pkg/cache/kubeConfig.go

```diff
@@ -0,0 +1,80 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jellydator/ttlcache/v3"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var kubeConfigCache = ttlcache.New[string, string](
+	ttlcache.WithTTL[string, string](5*time.Minute),
```

Comment:
Are 5min not too short? We reconcile with 20min intervals.. in theory we could increase the value to something <20min